streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -51,6 +51,16 @@
     abstract public void put(K key, V value);
 
     /**
+     * Update the value associated with this key, unless a value
+     * already is associated with the key
```

Comment:
Nitpick: `unless a value is already associated` reads better, I think. We need to document the return value since we have now added that. Also, we should make it clear that this may or may not be atomic (as people will expect it to be atomic given `ConcurrentMap.putIfAbsent`).